raster-foundry/granary
# Add file input component

## Overview

This PR adds a drag and drop + file selector file input for geojson.

Things that would probably be nice to have:

- an `x` to get rid of what you've chosen
- better errors about what's wrong if you upload `{"a": 3}` instead of an actual geojson

But they're not there. We can decide what would be nice to have while mobbing.

It also disables buttons instead of showing a hint (hi Alex). It also removes the feature flag thing from the PR template. It also separates the pages into modules so we can spend less time talking about file size while mobbing.

### Checklist

0 checklist items apply 🤷‍♂️ 

### Notes

Arguably login could also be moved to a module? I don't know. I feel like login as the top level view makes sense for the empty model case, which is what `Main.elm` starts with, so I like it where it is.

## Testing Instructions

- `./scripts/update --frontend`
- `docker-compose up -d database`
- navigate to `localhost:8080`
- enter a token from your database
- choose the Rasterio Water Mask model
- drag a random file onto the drop zone. It will show a nice orange border when you're in the right place.
- it should be mad at you about the file you uploaded
- drag a real geojson file onto the drop zone
- it should accept the file
- fill in the rest of the parameters with whatever you want and submit
- the API should also accept the geojson

Closes #394
